api/src/main/java/org/apache/iceberg/PartitionSpec.java

```diff
@@ -91,7 +94,7 @@ public int specId() {
    * @param fieldId a field id from the source schema
    * @return the {@link PartitionField field} that partitions the given source field
    */
-  public PartitionField getFieldBySourceId(int fieldId) {
+  public List<PartitionField> getFieldsBySourceId(int fieldId) {
```

Comment:
👍 I think this was actually the main issue - that this code assumes that there can be only one Partition field per field id, right?

Reply:
Yes, that's right.